src/types.ts

```diff
@@ -77,13 +77,29 @@ export type Dappeteer = {
      * Returns all notifications in Metamask notifications page
      */
     getAllNotifications: () => Promise<NotificationList>;
+    /** Invoke notification snap method. The function will open a notification
```

Comment:
Nope, we don't need this method. Please delete.
`getAllNotifications` and `waitForNotification` must be self-contained. User doesn't care about complex implementation but it cares about simple API.
You can create two internal functions:
- one will open new tab with metamask notifications screen
- other will close that tab

and in getAllNotifications and waitForNotification, you can use those methods to ensure you are on notifications screen but make sure after methods are done that tab is closed or main tab returned to home screen